src/NetAnalyzers/UnitTests/Microsoft.CodeQuality.Analyzers/QualityGuidelines/AvoidMultipleEnumerationsTests.cs

```diff
@@ -1284,7 +1283,28 @@ public void Sub(IEnumerable<int> h)
     }
 }";
             await VerifyCS.VerifyAnalyzerAsync(code);
+        }
+
+        [Fact]
+        public async Task TestInvocationLocalAssignmentWithDeferredMethodCall()
+        {
+            var code = @"
+using System;
+using System.Linq;
+using System.Collections;
+using System.Collections.Generic;
+
+public class Bar
```

Comment:
@mavasani I remember you were not a big fan of `Foobar`, I don't know if that's ok here.

Reply:
Should I change all the placeholder names here?